### PR TITLE
List contrib levels by official, community and user

### PR DESCRIPTION
**Pull request by mrkubax10**
This pull request adds feature from #720 and #1344

### CODE REVIEW

**Comment by tobbi** [comment]
Can you please rebase this? I think it's worth considering this.

File: src/supertux/menu/sorted_contrib_menu.hpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef HEADER_SUPERTUX_SUPERTUX_MENU_SORTED_CONTRIB_MENU_HPP
+#define HEADER_SUPERTUX_SUPERTUX_MENU_SORTED_CONTRIB_MENU_HPP
+#include "gui/menu.hpp"
+#include "supertux/world.hpp"
+class SortedContribMenu : public Menu
+{
+public:
+  SortedContribMenu(std::vector<std::unique_ptr<World>>& worlds, std::string contrib_type, std::string title, std::string empty_message);
```
Comment (Semphriss) [inline src/supertux/menu/sorted_contrib_menu.hpp:24]: `const std::string&`

File: src/supertux/menu/sorted_contrib_menu.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      }
+      else if (worlds[i]->is_worldmap())
+      {
+        m_world_folders.push_back(worlds[i]->get_basedir());
+        std::ostringstream title_stream;
+        title_stream << worlds[i]->get_title();
+        std::ostringstream desc;
+        desc << worlds[i]->get_description();
+        add_entry(world_id++, title_stream.str()).set_help(desc.str());
+      }
```
Comment (Semphriss) [inline src/supertux/menu/sorted_contrib_menu.cpp:57]: There's a lot of similarities in both parts of the `if`; isn't it possible to move some of it outside?

Possibly use the `?` operator if there isn't too many differences.

**Review by Semphriss — CHANGES_REQUESTED** [review]

File: src/supertux/menu/sorted_contrib_menu.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    {
+      m_world_folders.push_back(worlds[i]->get_basedir());
+      std::ostringstream title_stream;
+      if (worlds[i]->is_levelset())
+        title_stream << "[" << worlds[i]->get_title() << "]";
+      else
+        title_stream << worlds[i]->get_title();
+      std::ostringstream desc;
+      desc << worlds[i]->get_description();
+      add_entry(world_id++, title_stream.str()).set_help(desc.str());
```
Comment (Semphriss) [inline src/supertux/menu/sorted_contrib_menu.cpp:47]: These all seem to be strings - what are the string streams for?

**Review by Semphriss — CHANGES_REQUESTED** [review]

**Review by Semphriss — APPROVED** [review]

**Comment by Semphriss** [comment]
Extra idea if it sounds interesting: Given the category dedicated to the user's personal creations and the addition of the add-on creator in merged #1890, do you think it'd be feasible to populate the former with instances of the latter? E. g., fill the user contribs with the add-ons that the user packaged themselves.

**Comment by mrkubax10** [comment]
Originally I wanted to show user made levelsets in user category so I think it wouldn't work.

**Comment by tobbi** [comment]
Please rebase this so we can merge it!

**Pull request merged**